interfaces/cython/cantera/drawnetwork.py

```diff
@@ -0,0 +1,474 @@
+    node_attr = dict(kwargs.get("node_attr") or {}, **r.node_attr)
+
+    # include full reactor state in representation if desired
+    if print_state:
+        T_label = f"T (K)\\n{r.T:.2f}"
+        P_label = f"P (bar)\\n{r.thermo.P*1e-5:.3f}"
+        s_label = ""
+
+        if species == "X":
+            X = r.thermo.mole_fraction_dict(1e-4)
+            s_percents = "\\n".join([f"{s}: {v*100:.2f}" for s, v in X.items()])
+            s_label += "X (%)\\n" + s_percents
+        elif species == "Y":
+            Y = r.thermo.mass_fraction_dict(1e-4)
+            s_percents = "\\n".join([f"{s}: {v*100:.2f}" for s, v in Y.items()])
+            s_label += "Y (%)\\n" + s_percents
+        else:
+            # If individual species are provided as iterable of strings
+            if species:
+                X = r.thermo.mole_fraction_dict(threshold=-1)
+                s_percents = "\\n".join([f"{s}: {X[s]*100:.2f}" for s in species])
+                s_label += "X (%)\\n" + s_percents
+
+        # For full state output, shape must be 'Mrecord'
+        node_attr.pop("shape", None)
+        dot.node(_unique_name(r), shape="Mrecord",
+                 label=f"{{{T_label}|{P_label}}}|{s_label}",
+                 xlabel=r.name,
+                 **node_attr)
+
+    else:
+        dot.node(_unique_name(r), **node_attr)
+
+    return dot
+
+
+@_clear_reactor_names
+@_needs_graphviz
+def draw_reactor_net(n, *, print_state=False, species=None, graph_attr=None,
+                     node_attr=None, edge_attr=None, heat_flow_attr=None,
+                     mass_flow_attr=None):
+    """
+    Draw `ReactorNet` object as ``graphviz.graphs.DiGraph``. Connecting flow
+    controllers and walls are depicted as arrows.
+
+    :param n:
+        `ReactorNet` object
+    :param print_state:
+        Whether state information of the reactors is printed into each node.
+        Defaults to ``False``.
+    :param species:
+        If ``print_state`` is ``True``, define how species are to be printed.
+        Options are ``'X'`` and ``'Y'`` for mole and mass fractions of all
+        species, respectively, or an iterable that contains the desired species
+        names as strings. Defaults to ``None``.
+    :param graph_attr:
+        Attributes to be passed to the ``graphviz.Digraph`` function that
+        control the general appearance of the drawn network.
+        See https://graphviz.org/docs/graph/ for a list of all usable
+        attributes.
+    :param node_attr:
+        Attributes to be passed to the ``node`` method invoked to draw the
+        reactor. ``node_attr`` defined in the reactor object itself have
+        priority.
+        See https://graphviz.org/docs/nodes/ for a list of all usable
+        attributes.
+    :param edge_attr:
+        Attributes to be passed to the ``edge`` method invoked to draw
+        reactor connections. ``edge_attr`` defined in the connection objects
```

Comment:
In several locations:
```suggestion
        (subclasses of `FlowDevice` or walls) themselves have priority.
```

Reply:
Sorry, is this just about the missing "s" from "themselves" or were you suggesting additional changes here?

Reply:
Yes, I think that's all I was commenting on. Sorry for any confusion.